src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Fixer.cs

```diff
@@ -0,0 +1,201 @@
+        {
+            CancellationToken ct = context.CancellationToken;
+            if (ct.IsCancellationRequested)
+            {
+                return;
+            }
+
+            Document doc = context.Document;
+
+            SyntaxNode root = await doc.GetSyntaxRootAsync(ct).ConfigureAwait(false);
+
+            if (root.FindNode(context.Span, getInnermostNodeForTie: true) is not SyntaxNode node)
+            {
+                return;
+            }
+
+            SemanticModel model = await doc.GetSemanticModelAsync(ct).ConfigureAwait(false);
+
+            if (model.GetOperation(node, ct) is not IInvocationOperation invocation)
+            {
+                return;
+            }
+
+            if (model.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemStringComparison)
+                is not INamedTypeSymbol stringComparisonType)
+            {
+                return;
+            }
+
+            if (model.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemStringComparer)
+                is not INamedTypeSymbol stringComparerType)
+            {
+                return;
+            }
+
+            // Only diagnose overloads that take one string argument:
+            // Contains(string), IndexOf(string), StartsWith(string), CompareTo(string)
+            if (invocation.Arguments.Count() != 1)
+            {
+                return;
+            }
+
+            // There should be a child invocation on the left side (instance) of the large invocation
+            // If the large invocation is "a.ToLower().Contains(b)"
+            // Its instance is "a.ToLower()", the child invocation
+            if (invocation.Instance is not IInvocationOperation instanceOperation)
+            {
+                return;
+            }
+
+            string caseChangingMethodName;
+            switch (instanceOperation.TargetMethod.Name)
+            {
+                case RecommendCaseInsensitiveStringComparisonAnalyzer.StringToLowerMethodName or RecommendCaseInsensitiveStringComparisonAnalyzer.StringToUpperMethodName:
+                    caseChangingMethodName = RecommendCaseInsensitiveStringComparisonAnalyzer.StringComparisonCurrentCultureIgnoreCaseName;
+                    break;
+                case RecommendCaseInsensitiveStringComparisonAnalyzer.StringToLowerInvariantMethodName or RecommendCaseInsensitiveStringComparisonAnalyzer.StringToUpperInvariantMethodName:
+                    caseChangingMethodName = RecommendCaseInsensitiveStringComparisonAnalyzer.StringComparisonInvariantCultureIgnoreCaseName;
+                    break;
+                default:
+                    return;
+            }
+
+            Task<Document> createChangedDocument(CancellationToken _) => FixInvocationAsync(doc, root,
+                invocation, instanceOperation,
+                stringComparisonType, stringComparerType,
+                invocation.TargetMethod.Name, caseChangingMethodName, ct);
```

Comment:
`ct` doesn't feel right. I think we should use the CancellationToken provided to the lambda (which is currently discarded)